build_web_compilers/lib/src/dev_compiler_bootstrap.dart

```diff
@@ -108,10 +125,14 @@ Future<void> bootstrapDdc(BuildStep buildStep) async {
 final _lazyBuildPool = Pool(16);
 
 /// Ensures that all transitive js modules for [module] are available and built.
+///
+/// Throws an [UnsupportedModulesException] exception if there are any
+/// unsupported modules.
 Future<List<Module>> _ensureTransitiveModules(
     Module module, AssetReader reader) async {
   // Collect all the modules this module depends on, plus this module.
-  var transitiveDeps = await module.computeTransitiveDependencies(reader);
+  var transitiveDeps = await module.computeTransitiveDependencies(reader,
+      throwIfUnsupported: true);
```

Comment:
where is `computeTransitiveDependencies` used without `throwIfUnsupported`?

I wonder if we should change the check to not use exceptiosn?

```dart
var transitiveDeps = await module.computeTransitiveDependencies(reader);
var unsupported = transitiveDeps.where((d) => !d.isSupported).toList();
if (unsupported.isNotEmpty) {
  // log warning
}
```

Reply:
The main reason I use the exception is its a nice place to hang off utilities for finding the actual `ModuleLibrary` instances (the modules are not useful on their own really).

Reply:
Oh and also it makes it easier to handle errors at the top level and bail out - otherwise for instance this method can't make the overall builder bail out without making a more convoluted return type or returning null or something gross.